components/script/dom/document.rs

```diff
@@ -4196,149 +4196,112 @@ impl DocumentMethods for Document {
     }
 
     // https://html.spec.whatwg.org/multipage/#dom-document-open
-    fn Open(&self, _type: Option<DOMString>, replace: DOMString) -> Fallible<DomRoot<Document>> {
+    fn Open(&self, _unused1: Option<DOMString>, _unused2: Option<DOMString>) -> Fallible<DomRoot<Document>> {
+        // Step 1
         if !self.is_html_document() {
-            // Step 1.
             return Err(Error::InvalidState);
         }
 
-        // Step 2.
+        // Step 2
         if self.throw_on_dynamic_markup_insertion_counter.get() > 0 {
             return Err(Error::InvalidState);
         }
 
-        if !self.is_active() {
-            // Step 3.
-            return Ok(DomRoot::from_ref(self));
-        }
-
+        // Step 3
         let entry_responsible_document = GlobalScope::entry().as_window().Document();
 
+        // Step 4
         // This check is same-origin not same-origin-domain.
         // https://github.com/whatwg/html/issues/2282
         // https://github.com/whatwg/html/pull/2288
         if !self.origin.same_origin(&entry_responsible_document.origin) {
-            // Step 4.
             return Err(Error::Security);
         }
 
+        // Step 5
         if self
             .get_current_parser()
             .map_or(false, |parser| parser.is_active())
         {
-            // Step 5.
             return Ok(DomRoot::from_ref(self));
         }
 
-        // Step 6.
-        // TODO: ignore-opens-during-unload counter check.
-
-        // Step 7, 8.
-        // TODO: check session history's state.
-        let replace = replace.eq_ignore_ascii_case("replace");
-
-        // Step 9.
-        // TODO: salvageable flag.
-
-        // Step 10.
-        // TODO: prompt to unload.
+        // Step 6
+        if self.is_prompting_or_unloading() {
+            return Ok(DomRoot::from_ref(self));
+        }
 
         window_from_node(self).set_navigation_start();
 
-        // Step 11.
-        // TODO: unload.
-
-        // Step 12.
-        self.abort();
+        // Step 7
+        // TODO: Check if there is an existing attempt to navigate browsing context
+        if self.has_browsing_context() {
+            self.abort();
+        }
 
-        // Step 13.
+        // Step 8
         for node in self.upcast::<Node>().traverse_preorder() {
             node.upcast::<EventTarget>().remove_all_listeners();
         }
 
-        // Step 14.
-        // TODO: remove any tasks associated with the Document in any task source.
+        // Step 9
+        if let Some(window) = self.global().downcast::<Window>() {
+            if window.Document() == DomRoot::from_ref(self) {
+                self.global().upcast::<EventTarget>().remove_all_listeners()
+            }
+        }
 
-        // Step 15.
+        // Step 10.
         Node::replace_all(None, self.upcast::<Node>());
```

Comment:
This should avoid firing any mutation event, file an issue about this because I don't think we support it.